src/DGtal/dec/PolygonalCalculus.h

```diff
@@ -71,148 +71,183 @@ class PolygonalCalculus
+  /// normal.
+  /// @param surf an instance of SurfaceMesh
+  /// @param pos_embedder an embedder for the position
+  /// @param normal_embedder an embedder for the position
+  /// @param globalInternalCacheEnabled
+  PolygonalCalculus(const ConstAlias<MySurfaceMesh> surf,
+                    const std::function<Real3dPoint(Face,Vertex)> & pos_embedder,
+                    const std::function<Vector(Vertex)> & normal_embedder,
+                    bool globalInternalCacheEnabled = false) :
+  mySurfaceMesh(&surf), myEmbedder(pos_embedder), myVertexNormalEmbedder(normal_embedder), myGlobalCacheEnabled(globalInternalCacheEnabled)
+  {
+    init();
+  };
+
   /**
    * Deleted default constructor.
    */
   PolygonalCalculus() = delete;
-  
+
   /**
    * Destructor (default).
    */
   ~PolygonalCalculus() = default;
-  
+
   /**
    * Deleted copy constructor.
    * @param other the object to clone.
    */
-  PolygonalCalculus ( const PolygonalCalculus & other ) = delete;
-  
+  PolygonalCalculus(const PolygonalCalculus & other) = delete;
+
   /**
    * Deleted move constructor.
    * @param other the object to move.
    */
-  PolygonalCalculus ( PolygonalCalculus && other ) = delete;
-  
+  PolygonalCalculus(PolygonalCalculus && other) = delete;
+
   /**
    * Deleted copy assignment operator.
    * @param other the object to copy.
    * @return a reference on 'this'.
    */
-  PolygonalCalculus & operator= ( const PolygonalCalculus & other ) = delete;
-  
+  PolygonalCalculus & operator=(const PolygonalCalculus & other) = delete;
+
   /**
    * Deleted move assignment operator.
    * @param other the object to move.
    * @return a reference on 'this'.
    */
-  PolygonalCalculus & operator= ( PolygonalCalculus && other ) = delete;
+  PolygonalCalculus & operator=(PolygonalCalculus && other) = delete;
 
   /// @}
-  
+
   // ----------------------- embedding  services  --------------------------
   /// @name Embedding services
   /// @{
-  
+
   /// Update the embedding function.
   /// @param externalFunctor a new embedding functor (Face,Vertex)->RealPoint.
-  void setEmbedder(const std::function<Real3dPoint(Face,Vertex)> &externalFunctor)
+  void setEmbedder(const std::function<Real3dPoint(Face,Vertex)> & externalFunctor)
   {
     myEmbedder = externalFunctor;
   }
-  
-  // ----------------------- Per face operators --------------------------------------
+  /// @}
+
+  // ----------------------- Per face operators ------------------------------
   /// @name Per face operators
   /// @{
-  
+
   /// Return the vertex position matrix degree x 3 of the face.
   /// @param f a face
   /// @return the n_f x 3 position matrix
   DenseMatrix X(const Face f) const
   {
     if (checkCache(X_,f))
       return myGlobalCache[X_][f];
-    
-    const auto vertices = mySurfaceMesh->incidentVertices(f);
-    const auto nf = myFaceDegree[f];
+
+    auto vertices = mySurfaceMesh->incidentVertices(f);
```

Comment:
same here for the const / const auto stuff, it looks like you have edits the file without pulling the last version of PolygonalCalculus from the master.